src/spritefeat.asm

```diff
@@ -0,0 +1,336 @@
+;
+; Features using sprites to draw information
+;
+
+
+; Hook Gamemode 7 to upload tiles after loading if needed
+org $828B26
+    JSL update_sprite_tiles_loading
+
+; Hook sprite uploads in main game loop to get a chance to upload sprites first
+; Runs in Gamemode 8
+org $828B98
+    JSL update_sprite_features
+
+
+
+;
+; Add hitbox graphic to free sprite VRAM slot
+;
+org $9ADAE0
+db $FF, $FF, $80, $80, $80, $80, $80, $80, $80, $80, $80, $80, $80, $80, $80, $80
+db $FF, $FF, $80, $80, $80, $80, $80, $80, $80, $80, $80, $80, $80, $80, $80, $80 
+
+
+org $F0E000
+print pc, " spritefeat start"
+; This runs every frame before any other sprites are drawn, this is needed so we can get priority over everything else
+update_sprite_features:
+    PHA : PHX : PHY : PHP
+    
+    ; Draw OOB viewer if activated
+    JSR update_sprite_oob
+
+    ; Draw Samus hitbox if activated
+    JSR update_sprite_hitbox
+
+    PLP : PLY : PLX : PLA
+    
+    if !FEATURE_PAL
+        JSL $A0885D
+    else
+        JSL $A0884D
+    endif
+    
+    RTL
+
+; When loading into a game, do we need to update any sprite tiles?
+update_sprite_tiles_loading:
+    LDA !ram_oob_watch_active : BEQ +
+      JSL upload_sprite_oob_tiles
+    +
+    
+    JSL $80894D
+    RTL
+
+upload_sprite_oob_tiles:
+    LDX $330
+    LDA #$0400
+    STA $D0, X
+    LDA #(sprite_tiles+$200)
+    STA $D2, X
+    LDA #((sprite_tiles+$200)>>8)
+    STA $D3, X
+    LDA #$6D00
+    STA $D5, X
+    TXA : CLC : ADC #$0007 : STA $0330
+    RTL
+
+update_sprite_oob:
+    LDA !ram_oob_watch_active : BNE +
+    RTS
++
+    !oob_width = $000D
+    !oob_height = $0009
+    LDA $590 : STA $C8
+
+    LDA $0AF6
+    SEC : SBC #(!oob_width*8)
```

Comment:
Thank you for this example of local constants.  This will make it easier for me to update the shinetune method so it works for PAL.